src/google_map.js

```diff
@@ -453,9 +454,10 @@ export default class GoogleMap extends Component {
   _mapDomResizeCallback = () => {
     this.resetSizeOnIdle_ = true;
     if (this.maps_) {
-      const originalCenter = this.map_.getCenter();
+      const originalCenter = this.props.center || this.props.defaultCenter;
+      const currentCenter = this.map_.getCenter();
       this.maps_.event.trigger(this.map_, 'resize');
-      this.map_.setCenter(originalCenter);
+      this.map_.setCenter(this.props.lockCenter ? originalCenter : currentCenter);
```

Comment:
@Marcel-G This PR is looking good! But it is throwing an error in this line on Travis CI:

`460:27  error  Follow `prettier` formatting (expected '\n' but found 't')  prettier/prettier`

Would you mind checking on that?

Reply:
@itsmichaeldiego, thanks for stepping up as a new maintainer! I've already upgraded to the latest version, thanks to the fix in https://github.com/istarkov/google-map-react/pull/452#issuecomment-353203783. However, I need to get this fix landed as well before I can actually take advantage of the fullscreen functionality.

The CI error can be fixed by running `yarn lint --fix`, or applying the following patch:

```diff
diff --git a/src/google_map.js b/src/google_map.js
index aba1ea5..e4a8b8c 100644
--- a/src/google_map.js
+++ b/src/google_map.js
@@ -457,7 +457,9 @@ export default class GoogleMap extends Component {
       const originalCenter = this.props.center || this.props.defaultCenter;
       const currentCenter = this.map_.getCenter();
       this.maps_.event.trigger(this.map_, 'resize');
-      this.map_.setCenter(this.props.lockCenter ? originalCenter : currentCenter);
+      this.map_.setCenter(
+        this.props.lockCenter ? originalCenter : currentCenter
+      );
     }
   };
 
```

Would you like me to submit a separate PR?

Reply:
@josephfrazier Thanks for the kind words. This fix seems like the solution to our problems. Feel free to make a PR, that would be awesome! We would also need a modification in the docs so users know they're able to use `lockCenter`

Reply:
Thanks for the quick feedback, see https://github.com/istarkov/google-map-react/pull/482